devlog/_plan/260816_wave34_closeout/030_1789_workspace_outcome.md

```
@@ -0,0 +1,54 @@
+type CodexUpstreamEvidence = {
+  status: number | "connect_error" | "timeout" | "connect_neutral";
+  /** Set when the upstream body identified a workspace/entitlement denial rather than a bad credential. */
+  denial?: "workspace" | "entitlement";
+};
+
+classifyCodexUpstreamOutcome(evidence: CodexUpstreamEvidence): CodexUpstreamOutcomeClass
+```
+
+   401 stays `credential`. A 403 with `denial: "workspace"` becomes `workspace`; a 403 with no denial evidence stays `credential`, so the change fails safe toward today's behavior.
+
+3. **Carry it on the existing meta rather than changing every signature.** `CodexUpstreamOutcomeMeta` already reaches `recordCodexUpstreamOutcome` from every call site, so adding `denial?: "workspace" | "entitlement"` there means only the sites that can actually observe a 403 body need to populate it — in the Responses path, the two `quotaMeta` construction points.
```

markdownlint-cli2 (0.23.2)

[warning] Ordered list item prefix
Expected: 1; Actual: 3; Style: 1/2/3

(MD029, ol-prefix)